xds/internal/client/client_xds.go

```diff
@@ -181,8 +177,14 @@ func processServerSideListener(lis *v3listenerpb.Listener, logger *grpclog.Prefi
 }
 
 func getAddressFromName(name string) (host string, port string, err error) {
+	index := strings.Index(name, "udpa.resource.listening_address=")
+	if index == -1 {
+		return "", "", fmt.Errorf("udpa.resource_listening_address not found in name: %v", name)
+	}
+	lisAddr := name[index:]
+
 	var addr string
-	_, err = fmt.Sscanf(name, "grpc/server?udpa.resource.listening_address=%s", &addr)
+	_, err = fmt.Sscanf(lisAddr, "udpa.resource.listening_address=%s", &addr)
```

Comment:
Can you do `fmt.Sscanf(name, "%s?udpa.resource.listening_address=%s", &type /* or &dummy since you don't seem to need it */, &addr)`?

Reply:
I had tried this as well. But the first "%s" gets the full string :(

Reply:
Slightly simpler, then:

```go
	parts := strings.SplitN(name, "udpa.resource.listening_address=", 2)
	if len(parts) != 2 {
		return "", "", fmt.Errorf("udpa.resource_listening_address not found in name: %v", name)
	}
	addr := parts[1]
```

Or you can keep the `Index` but just use `addr := name[index+len("udpa.resource.listening_address="):]` instead of the `Sscanf`

Reply:
Done. Thanks.